activesupport/lib/active_support/testing/time_helpers.rb

```diff
@@ -30,13 +38,27 @@ def unstub_all!
       end
 
       private
-
         def unstub_object(stub)
           singleton_class = stub.object.singleton_class
           singleton_class.send :undef_method, stub.method_name
           singleton_class.send :alias_method, stub.method_name, stub.original_method
           singleton_class.send :undef_method, stub.original_method
         end
+
+        def store_existing_stub(object, method_name)
+          key = [object.object_id, method_name]
+
+          if stub = @stubs[key]
+            old_stubs = @pre_stubs.delete(key){ Array.new }
+            old_stubs << {
+              object: object,
+              method_name: method_name,
+              return_value: object.send(method_name)
+            }
+            @pre_stubs.merge!(key => old_stubs)
```

Comment:
Why are we just deleting and then merging? Why not just `@pre_stubs[key] ||= []; @pre_stubs[key] << ...`